o2a/mappers/fs_mapper.py

```diff
@@ -112,47 +116,85 @@ class FsMapper(ActionMapper):
     Converts a FS Oozie node to an Airflow task.
     """
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(
+        self,
+        oozie_node: Element,
+        name: str,
+        dag_name: str,
+        job_properties: Dict[str, str],
+        configuration_properties: Dict[str, str],
+        trigger_rule: str = TriggerRule.ALL_SUCCESS,
+        **kwargs,
+    ):
+        super().__init__(
+            oozie_node=oozie_node,
+            name=name,
+            job_properties=job_properties,
+            configuration_properties=configuration_properties,
+            trigger_rule=trigger_rule,
+            dag_name=dag_name,
+            **kwargs,
+        )
+        self.oozie_node = oozie_node
+        self.trigger_rule = trigger_rule
         self.tasks: List[Task] = []
+        self.action_node_properties_to_add: Optional[Dict[str, str]] = None
 
     def on_parse_node(self):
         super().on_parse_node()
         self.tasks = self.parse_tasks()
 
-    def parse_tasks(self):
+    def parse_tasks(self) -> List[Task]:
         if not list(self.oozie_node):
             return [Task(task_id=self.name, template_name="dummy.tpl", trigger_rule=self.trigger_rule)]
 
-        return [self.parse_fs_action(i, node) for i, node in enumerate(self.oozie_node)]
-
-    def to_tasks_and_relations(self):
+        tasks: List[Task] = []
+        index = 0
+        for node in self.oozie_node:
+            task = self.parse_fs_action(index, node)
+            if task is not None:
+                tasks.append(task)
+                index += 1
+            else:
+                # configuration node needs to be skipped
+                pass
+        return tasks
+
+    def to_tasks_and_relations(self) -> Tuple[List[Task], List[Relation]]:
         return self.tasks, chain(self.tasks)
 
     def required_imports(self) -> Set[str]:
-        return {
-            "import shlex",
-            "from airflow.operators import dummy_operator",
-            "from airflow.operators import bash_operator",
-        }
+        return {"from airflow.operators import dummy_operator", "from airflow.operators import bash_operator"}
 
     @property
-    def first_task_id(self):
+    def first_task_id(self) -> str:
         return self.tasks[0].task_id
 
     @property
-    def last_task_id(self):
+    def last_task_id(self) -> str:
         return self.tasks[-1].task_id
 
-    def parse_fs_action(self, index: int, node: Element):
+    def parse_fs_action(self, index: int, node: Element) -> Optional[Task]:
         tag_name = node.tag
+        if tag_name == "configuration":
```

Comment:
I have a different idea. I think we should omit the nodes here, which are not supported. ie. parse only supset of nodes
```
            <xs:choice minOccurs="0" maxOccurs="unbounded">
                <xs:element name="delete" type="workflow:DELETE"/>
                <xs:element name="mkdir" type="workflow:MKDIR"/>
                <xs:element name="move" type="workflow:MOVE"/>
                <xs:element name="chmod" type="workflow:CHMOD"/>
                <xs:element name="touchz" type="workflow:TOUCHZ"/>
                <xs:element name="chgrp" type="workflow:CHGRP"/>
            </xs:choice>
```
Therefore, the configuration will normally be supported by ActionMapper. 

ActionMapper should have a code similar to this:
```
class ActionMapper(BaseMapper):
    """Base class for all action mappers"""

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.properties: Dict[str, str] = {}

    def on_parse_node(self, workflow):
        super().on_parse_node(workflow)
        self._parse_configuration(workflow)
```

Reply:
👍 Fixed in ccc01c1853457a0bec5ab0cef8e088b5b2d4ae54